battle-engine.js

```diff
@@ -492,6 +492,56 @@ BattlePokemon = (function () {
 		}
 		return null;
 	};
+	BattlePokemon.prototype.getMoveTargets = function (move, target) {
+		move = this.battle.getMove(move);
+		var targets = [];
+		switch (move.target) {
+		case 'all':
+		case 'foeSide':
+		case 'allySide':
+		case 'allyTeam':
+			if (move.target.substr(0, 3) !== 'foe') {
```

Comment:
`.slice` is for diving strings up into slices, usually at indexes you get from `.indexOf`. It's a more controllable `.split` and also useful in performance-intensive code.

`.substr` is for grabbing substrings of specific lengths from strings.

`.substring` is basically never useful, it's just a `.slice` with fewer features, and it can easily be misread for `.substr`.

Basically, if you think "from A to B", use `.slice`, and if you think "from A for B characters", use `.substr`.